modules/pipeline/providers/clusterinfo/interface.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusterinfo
+
+import (
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/bundle"
+	"github.com/erda-project/erda/pkg/discover"
+	"github.com/erda-project/erda/pkg/strutil"
+)
+
+var (
+	ClusterHookApiPath = "/api/pipeline-clusters/actions/hook"
+)
+
+// Interface defines the interface for the clusterinfo provider
+type Interface interface {
+	// GetClusterByName return ClusterInfo query by name, contain cluster connection info and config map data
+	GetClusterInfoByName(string) (apistructs.ClusterInfo, error)
+	// GetClusterInfoCMByName return config map data of cluster query by name, data is displayed as map
+	GetClusterInfoCMByName(string) (map[string]string, error)
```

Comment:
merge `GetClusterInfoByName` and `GetClusterInfoCMByName` into one.

Reply:
adjusted